lib/matplotlib/tests/test_contour.py

```diff
@@ -171,6 +171,18 @@ def test_given_colors_levels_and_extends():
         plt.colorbar(c, ax=ax)
 
 
+@pytest.mark.parametrize('color', ['green', ('r', 0.5), (0.1, 0.2, 0.5, 0.3)])
+@pytest.mark.parametrize('extend', ['neither', 'both'])
+@pytest.mark.parametrize('levels', [[0.5, 1, 1.5], [0.5, 0.75, 1, 1.25, 1.5]])
```

Comment:
Why do we need `levels` parametrization? I would not expect that to have an impact now that we accept all color specs. Due to the generalization to arbitrary color specs, every "reasonable" implementation must first detect single colors and normalize to a list of colors. This should exclude the previous special behavior when color strings match in length with levels. Note also that you test three sequence-like single colors, so you already can check both cases, these having same or different lengths with `levels` with one fixed set of levels.

Generally, I'd try to avoid multiple parametrizations when they are basically independent, because the number of tests in the product space increases quickly. Here you create 12 tests, when in fact you only 4-5

- Three colors, to test different color specs (extent=neigher)
- one additional test for the  extent=both
- optionally one additional test for another levels setting

Reply:
Fair point.  How's [this](https://github.com/matplotlib/matplotlib/compare/5ff1f17c4fe21bafab5ca617f9c57415dd21f258..8920c62737450e0d4eb85414cbed931f6aa92e1f)?  The point about "green" is that it is two characters longer than the number of levels and ends with a character that is not itself a colour.  So I picked "darkred" instead.  More levels were needed to produce the failure with the RBGA case.